lib/src/excel.dart

```diff
@@ -76,66 +80,98 @@ class ExcelDart {
+  /// If [_tableHeader] has been to changed, the offset here needs to
+  /// be changed to match too.
+  static void _formatBudgetColumn(String tableName) async {
+    await ExcelJS.run(allowInterop((context) async {
+      final table = context.workbook.tables.getItem(tableName);
+      final budgetRange = table.columns.getItem('Budget').getRange();
+      final format = budgetRange.conditionalFormats.add('Custom');
+
+      format.custom.rule.formula = '=IF(INDIRECT("RC[-2]",0) '
+          '="${InsertionOrder_InsertionOrderBudget_BudgetUnit.BUDGET_UNIT_CURRENCY}",'
+          'TRUE)';
+      format.custom.format.numberFormat = _currencyFormat;
+
+      await context.sync();
     }));
   }
 
-  static List<String> _generateTableRow(InsertionOrder io) => [
-        io.insertionOrderId,
-        io.advertiserId,
-        io.campaignId,
-        io.displayName,
-        io.entityStatus.toString(),
-        io.updateTime,
-        io.pacing.pacingPeriod.toString(),
-        io.pacing.pacingType.toString(),
-        _calculatePacingDailyMax(io.pacing.dailyMaxMicros),
-        io.pacing.dailyMaxImpressions,
-        io.budget.budgetUnit.toString(),
-        io.budget.automationType.toString(),
-        _calculateTotalBudgetAmount(io.budget.budgetSegments),
-        _calculateStartDate(io.budget.budgetSegments.first),
-        _calculateEndDate(io.budget.budgetSegments.last),
-      ];
+  /// Generates a row from [insertionOrder] by creating fields that matches
+  /// those specified in [_tableHeader]
+  static List<String> _generateTableRow(InsertionOrder insertionOrder) {
+    final row = [
+      insertionOrder.insertionOrderId,
+      insertionOrder.advertiserId,
+      insertionOrder.campaignId,
+      insertionOrder.displayName,
+      insertionOrder.entityStatus.toString(),
+      insertionOrder.updateTime,
+      insertionOrder.pacing.pacingPeriod.toString(),
+      insertionOrder.pacing.pacingType.toString(),
+      _calculatePacingDailyMax(insertionOrder.pacing.dailyMaxMicros),
+      insertionOrder.pacing.dailyMaxImpressions
+    ];
+
+    row.addAll(_generateBudgetColumns(insertionOrder.budget));
+    return row;
+  }
 
+  /// Calculates pacing daily max in standard unit.
   static String _calculatePacingDailyMax(String dailyMaxMicros) =>
       dailyMaxMicros.isEmpty
           ? dailyMaxMicros
           : (Int64.parseInt(dailyMaxMicros) * 1e-6).toString();
 
-  /// Todo: calculate budget total for active segments only.
-  /// Issue: https://github.com/googleinterns/dv360-excel-plugin/issues/35.
-  static String _calculateTotalBudgetAmount(
-      List<InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment>
-          budgetSegments) {
-    final totalBudgetMicros = budgetSegments
+  /// Generates columns for budget unit, automation type,
+  /// active segments total budget, start date and end date from [budget]
+  static List<String> _generateBudgetColumns(
```

Comment:
why combine these all into one method vs having them split out as before?

Reply:
Hmmm...nothing really, but now I look at it and agree that splitting them out is more readable.